expression/constraint_propagation.go

```diff
@@ -34,14 +35,14 @@ type exprSet struct {
 	constfalse bool
 }
 
-func (s *exprSet) Append(e Expression) bool {
-	if _, ok := s.exists[string(e.HashCode(nil))]; ok {
+func (s *exprSet) Append(sc *stmtctx.StatementContext, e Expression) bool {
```

Comment:
This `StatementContext` maybe not the same one with the original `StatementContext` which is used in the create table DDL.
But I haven't find a better way to handle it yet.